### PR TITLE
Add cmd to get the score for a whole batch at once

### PR DESCRIPTION
**Pull request by aseem09**
Our bot already has a command `bot score` which tells us the `score` of the user, i.e., the amount of appreciation received. But let's say we want to know the score of a whole batch, we'll need to manually
enter the name of each member, which is quite a tedious task. Add `batchwise-score` script to make this task easier. Now the user will be able to get the score of each user in a particular batch all at once by the command `bot score bxx`

### CODE REVIEW

File: scripts/info.coffee
Comment (r1walz) [inline scripts/info.coffee]: @aseem09 I like the effort to improve the functionality of the bot. But this and the other commit are not logically related. If you want to add this `spam` filter then please submit it as an independent PR.

File: scripts/batch-score.coffee
Comment (r1walz) [inline scripts/batch-score.coffee:1]: A note on commit message: It provides no information at all. @aseem09 could you change it to something which tells us `why` we are doing `what` we are doing? Something like:
```
scripts: add batch-score.coffee

Our bot already has a command `bot score` which tells us the `score`
of the user, i.e., the amount of appreciation received. But let's say
we want to know the score of a whole batch, we'll need to manually
enter the name of each member, which is quite a tedious task. Add
`batch-score` script to make this task easier.
```

Tweak this message if you like :wink:.
NB: the message is wrapped at around 72 chars, which makes `git log` to indent it beautifully.

File: scripts/batch-score.coffee
Comment (r1walz) [inline scripts/batch-score.coffee:2]: `batch-wise` may be?

**Comment by r1walz** [comment]
@aseem09 Thanks for the PR. This looks good. Please consider the changes that I've suggested. Also, I've not reviewed the code since I am running busy lately. @dev-ritik Can you take it from here? I'll try to review it ASAP.

File: scripts/info.coffee
Comment (r1walz) [inline scripts/info.coffee]: there is no need for spaces around `\n`

File: scripts/batch-score.coffee
Comment (r1walz) [inline scripts/batch-score.coffee:1]: > Our bot already has a command bot score which tells us the score of the user, i.e., the amount of appreciation received. But let's say we want to know the score of a whole batch, we'll need to manually
enter the name of each member, which is quite a tedious task. Add batchwise-score script to make this task easier. Now the user will be able to get the score of each user in a particular batch all at once by the command bot score bxx

Some comments on commit message:
1. Your commit message is not wrapped around 72 characters.
2. You've added batch-score.coffee not batchwise-score.coffee. I prefer batch-score.coffee since we'll be adding some more commands and considering them `batch-score` is a better name.

> Now the user will be able to get the score of each user in a particular batch all at once by the command bot score bxx

3. This line is unnecessary since it is more of a "documentation" line. If you want you can add it to the script itself but I don't suggest so. We want to include "why" we need this file. Not "how" to use it or "what" it does.

Comment (aseem09) [reply scripts/batch-score.coffee:1]: I have done the necessary changes. Please check.

Comment (r1walz) [reply scripts/batch-score.coffee:1]: Don't commit just now. Wait for @dev-ritik to review this PR.

Comment (r1walz) [reply scripts/batch-score.coffee:1]: another nit: your oneliner starts with `Adds` change it to `Add`. We should write messages as if we are instructing the code base to do as said.

Comment (r1walz) [reply scripts/batch-score.coffee:1]: nit: And get this through your mind. **Every line of the commit message should be less than 72 characters**

File: scripts/batch-score.coffee
Comment (r1walz) [inline scripts/batch-score.coffee]: `b[\w\-_]` matches any line starting with `b`, containing only alphanumeric chars thereafter (also - and _). So, I think `b\d\d` could be a better regex since it'll match `b<number><number>` (eg b21, B22).

**Review by dev-ritik — CHANGES_REQUESTED** [review]
I have added some reviews. Particularly the algorithm needs to be changed!

File: scripts/batch-score.coffee
Comment (dev-ritik) [inline scripts/batch-score.coffee]: Are you sure it will be consistent with your data?

File: scripts/batch-score.coffee
Comment (dev-ritik) [inline scripts/batch-score.coffee]: This method to get the year is very fragile! It will cause mapping errors after we restart the script! Also fixing a particular date will create confusion among the users

File: scripts/batch-score.coffee
Comment (dev-ritik) [inline scripts/batch-score.coffee]: Why do you need `msgcount` of the users?

File: scripts/batch-score.coffee
Comment (aseem09) [reply scripts/batch-score.coffee]: I have tested this by hard coding temp2 as <digit><alphabet><alphabet> eg. "1st"

File: scripts/batch-score.coffee
Comment (aseem09) [reply scripts/batch-score.coffee]: Forgot to change it. I'll do the necessary changes

File: scripts/batch-score.coffee
Comment (aseem09) [reply scripts/batch-score.coffee]: I have done some changes. Please have a look

File: scripts/batch-score.coffee
Comment (aseem09) [reply scripts/batch-score.coffee]: Done.

File: scripts/batch-score.coffee
Comment (r1walz) [inline scripts/batch-score.coffee]: drop "using this": `get the data of brain's scorefield key` is just fine. I think we can drop the comment altogether.
Tip: write code that speaks for itself, i.e., that need not be commented.

File: scripts/batch-score.coffee
Comment (r1walz) [inline scripts/batch-score.coffee]: We already know that this script will generate some response. Better explain "how" control flows within the functions.

File: scripts/batch-score.coffee
Comment (r1walz) [inline scripts/batch-score.coffee]: Can we do something like this instead?
```coffee
  if result != ""
    result
  false
```

If I am not wrong:

```js
  if result != ""
    result
```
if equivalent to:

```js
  if result != ""
    return result
```

File: scripts/batch-score.coffee
Comment (r1walz) [inline scripts/batch-score.coffee]: I don't' know why do we need '('s around `\d`s. Would you care to explain?

File: scripts/batch-score.coffee
Comment (r1walz) [inline scripts/batch-score.coffee]: Can we have meaningful names instead? Like this is an array storing `scores`.

File: scripts/batch-score.coffee
Comment (r1walz) [inline scripts/batch-score.coffee]: Can't we just reuse temp5 here? (again change it to some meaningful name)

File: scripts/batch-score.coffee
Comment (r1walz) [inline scripts/batch-score.coffee]: This comment gives no more information than it is obvious, hence drop it.

File: scripts/batch-score.coffee
Comment (r1walz) [inline scripts/batch-score.coffee]: Can be simplified to `String(str).split('').length`. Many web dev languages have `strlen` to calculate the length of a string. We may want to change this to `strlen` if coffee already does not have any method to achieve so.

File: scripts/batch-score.coffee
Comment (r1walz) [inline scripts/batch-score.coffee]: May add a comment that tells what `pad` is doing here, i.e., it makes every element the same "width".

File: scripts/batch-score.coffee
Comment (r1walz) [inline scripts/batch-score.coffee]: again use meaningful names

File: scripts/batch-score.coffee
Comment (r1walz) [inline scripts/batch-score.coffee]: It is a common practice to declare your functions before you use. Although this will not raise any error(s) but it's a nice practice to follow since people reading your code may stumble if they come across a function which is not in `stdlib` or not yet declared.

**Review by r1walz — CHANGES_REQUESTED** [review]
State of PR looks better. Some comments before it's ready to be merged. Thanks for your work!

File: scripts/batch-score.coffee
Comment (aseem09) [reply scripts/batch-score.coffee]: I tried. It doesnt work without the braces!

**Review by dev-ritik — APPROVED** [review]

File: scripts/batch-score.coffee
Comment (dev-ritik) [inline scripts/batch-score.coffee]: These 3 statements can be simplified to one statement.

File: scripts/batch-score.coffee
Comment (r1walz) [inline scripts/batch-score.coffee:20]:
```suggestion
  stringLength = (str) ->
    return String(str).split('').length
```

File: scripts/batch-score.coffee
Comment (r1walz) [inline scripts/batch-score.coffee]: I know we don't have any styling guide ... yet. But let's follow some of the standard styles till then?
Some guidelines:
1. There should be spaces around `=`
2. Any line should at most be of 80 chars length
3. `then` should come on the next line of `if` (same for `else`)
4. Avoid global variables

Comment (aseem09) [reply scripts/batch-score.coffee]: Okay

**Comment by r1walz** [comment]
Thanks! That was a pleasant contribution.

**Pull request merged**
